pkg/agent/agent.go

```diff
@@ -312,78 +312,36 @@ func (i *Initializer) initInterfaceStore() error {
 		var intf *interfacestore.InterfaceConfig
 		interfaceType, ok := port.ExternalIDs[interfacestore.AntreaInterfaceTypeKey]
 		if !ok {
-			interfaceType = interfacestore.AntreaUnset
-		}
-		if interfaceType != interfacestore.AntreaUnset {
-			switch interfaceType {
-			case interfacestore.AntreaGateway:
-				intf = parseGatewayInterfaceFunc(port, ovsPort)
-			case interfacestore.AntreaUplink:
-				intf = parseUplinkInterfaceFunc(port, ovsPort)
-			case interfacestore.AntreaTunnel:
-				fallthrough
-			case interfacestore.AntreaIPsecTunnel:
-				intf = parseTunnelInterfaceFunc(port, ovsPort)
-			case interfacestore.AntreaHost:
-				if port.Name == i.ovsBridge {
-					// Need not to load the OVS bridge port to the interfaceStore
-					intf = nil
-				} else {
-					var err error
-					intf, err = externalnode.ParseHostInterfaceConfig(i.ovsBridgeClient, port, ovsPort)
-					if err != nil {
-						return fmt.Errorf("failed to get interfaceConfig by port %s: %v", port.Name, err)
-					}
-				}
-			case interfacestore.AntreaContainer:
-				// The port should be for a container interface.
-				intf = cniserver.ParseOVSPortInterfaceConfig(port, ovsPort)
-			case interfacestore.AntreaTrafficControl:
-				intf = trafficcontrol.ParseTrafficControlInterfaceConfig(port, ovsPort)
-			default:
-				klog.InfoS("Unknown Antrea interface type", "type", interfaceType)
-			}
-		} else {
-			// Antrea Interface type is not saved in OVS port external_ids in earlier Antrea versions, so we use
-			// the old way to decide the interface type for the upgrade case.
-			uplinkIfName := i.nodeConfig.UplinkNetConfig.Name
-			var antreaIFType string
-			switch {
-			case port.OFPort == config.HostGatewayOFPort:
-				intf = parseGatewayInterfaceFunc(port, ovsPort)
-				antreaIFType = interfacestore.AntreaGateway
```

Comment:
With this change and the v2 release, I think it makes sense to drop support for this legacy scenario. The interface type externalID was introduced by @wenyingd starting with Antrea v1.5 (#3027).
 
With the change to the default value of `HostGatewayOFPort`, this switch case (`port.OFPort == config.HostGatewayOFPort`) could no longer be supported.

cc @tnqn

Reply:
Sounds good to me